### PR TITLE
Add checklist for teaching deployment

### PR DESCRIPTION
**Pull request by willingc**
This PR is a first pass at putting together a checklist to help guide developers through the deployment of JupyterHub for teaching. The goal is to help the person deploying to avoid missing a necessary step or configuration element.

### CODE REVIEW

File: docs/source/teaching-checklist.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+- Single Ubuntu server
+- [Ansible](https://www.ansible.com/resources) for configuration.
+
+## 1. Prepare the server
+
+- [ ] **Server:** running latest Ubuntu version
+- [ ] **SSH:** enable password-less SSH for `ubuntu` user
+- [ ] **Local disks:** partition and format
+- [ ] **DNS (domain name):** valid entry for server
+- [ ] **SSL certificate/key:** obtain via let's encrypt or trusted source
```
Comment (minrk) [inline docs/source/teaching-checklist.md]: The letsencrypt step is done by ansible, so it should be clear that you don't need to have SSL setup before you start if using letsencrypt. I see this is noted in step 3, but I don't want people to get held up here.

File: docs/source/teaching-checklist.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+- [ ] **`./hosts` file:** Edit file to lists the FQDN's of the hosts in the
+  `jupyterhub_hosts` group.
+- [ ] **hostname files:**  Use `./host_vars/hostname.example` as a
+  template for creating and editing a hostname file for each host and
+  place hostname files in `./host_vars` directory.
+
+## 5. Configure admins
+
+- [ ] List?
+- [ ] Public SSH keys
```
Comment (minrk) [inline docs/source/teaching-checklist.md]: Configured in `jupyterhub_admin_users` in config file. Public ssh keys are retrieved from GitHub.

File: docs/source/teaching-checklist.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+## 6. Configure users
+
+- [ ] If using [PAM (Pluggable authentication
+  modules)](https://en.wikipedia.org/wiki/Linux_PAM, you will need to
+  manually create users using adduser: `adduser --gecos "" username`.
+
+- [ ] If using [GitHub OAuth](https://developer.github.com/v3/oauth/):
```
Comment (minrk) [inline docs/source/teaching-checklist.md]: ... add usernames to `jupyterhub_users` list.

File: docs/source/teaching-checklist.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+## 7. Add optional services
+
+- [ ] **Monitoring:** New Relic
```
Comment (minrk) [inline docs/source/teaching-checklist.md:68]: Also Google Analytics

File: docs/source/teaching-checklist.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+    ```bash
+    $ ssh root@{hostname}
+    ```
+    substituting your hostname for {hostname}. For example, ``ssh root@jupyter.org``.
+
+- [ ] **supervisor:** Reload.
+
+    ```bash
+    $ supervisorctl reload
```
Comment (minrk) [inline docs/source/teaching-checklist.md]: This step shouldn't be needed anymore.

**Review by minrk — COMMENTED** [review]
Great, thanks @willingc! I made a few comments inline

**Comment by willingc** [comment]
@minrk I think that I've made all the requested changes.

**Pull request merged**